internal/controller/bucket/helpers.go

```diff
@@ -0,0 +1,70 @@
+package bucket
+
+import (
+	"context"
+
+	s3types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/linode/provider-ceph/apis/provider-ceph/v1alpha1"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+)
+
+// isAlreadyExists helper function to test for ErrCodeBucketAlreadyOwnedByYou error
+func isAlreadyExists(err error) bool {
+	var alreadyOwnedByYou *s3types.BucketAlreadyOwnedByYou
+
+	return errors.As(err, &alreadyOwnedByYou)
+}
+
+func setBucketStatus(bucket *v1alpha1.Bucket, bucketBackends *bucketBackends) {
+	bucket.Status.SetConditions(xpv1.Unavailable())
+
+	bucketBackendStatuses := bucketBackends.getBucketBackendStatuses(bucket.Name, bucket.Spec.Providers)
+	bucket.Status.AtProvider.BackendStatuses = bucketBackendStatuses
+
+	for _, backendStatus := range bucketBackendStatuses {
+		if backendStatus == v1alpha1.BackendReadyStatus {
+			bucket.Status.SetConditions(xpv1.Available())
+
+			break
+		}
+	}
+}
+
+// Callbacks have to return false on object update and true on status update.
```

Comment:
Instead of returning `bool`, could we have the callbacks return something like...

```
type UpdateRequired int

const (
	NeedsStatusUpdate UpdateRequired = iota
	NeedsObjectUpdate
)
```
I think it would make the calls from `Update()`, `Create()` and `Delete()` more readable